### ISSUE TITLE
Coverage 6.0 moved `coverage.misc.NoSource` to `coverage.exceptions.NoSource`

### ISSUE THREAD

**Issue by haakenlid** [TheKevJames/coveralls-python#326]
Looks like the latest stable version of coverage breaks the reporting from coveralls-python.

```
#!/bin/bash -eo pipefail
COVERALLS_REPO_TOKEN=Q58WdUuZOi89XHyDeDsGE2lxUGQ2IfqP3 coveralls
Traceback (most recent call last):
  File "/home/circleci/.local/bin/coveralls", line 5, in <module>
    from coveralls.cli import main
  File "/home/circleci/.local/lib/python3.8/site-packages/coveralls/__init__.py", line 1, in <module>
    from .api import Coveralls
  File "/home/circleci/.local/lib/python3.8/site-packages/coveralls/api.py", line 14, in <module>
    from .reporter import CoverallReporter
  File "/home/circleci/.local/lib/python3.8/site-packages/coveralls/reporter.py", line 5, in <module>
    from coverage.misc import NoSource
ImportError: cannot import name 'NoSource' from 'coverage.misc' (/home/circleci/.local/lib/python3.8/site-packages/coverage/misc.py)

Exited with code exit status 1
CircleCI received exit code 1
```

https://github.com/nedbat/coveragepy/commit/6a3d3aaaf2aebb816c7287263c8097844280b233#diff-8489ba0fa1175208cbaa2e586bbf94c70fe721b8ebd8f022452e431fefd64168

**Comment by nedbat** [comment]
I took a look at reporter.py here: there are *many* internal elements being accessed.  It would be better to work out a public API that does what you need.

**Comment by haakenlid** [comment]
I did figure out a solution for my problem. Turns out the circleCI configuration of the github repo I was submitting a pull request to would istall the latest stable version of both coveralls and coverage python packages.

    steps:
      - checkout
      - run:
          command: pip install --user tox coveralls
          name: "Install Tox & Coverage"

I changed this to let covaralls choose which version of coverage to use, and then it built successfully.

    command: pip install --user tox coveralls
    
I suspect that there are more projects out there with similar configuration (copy-pasted?) that will be affected by this change.
